libs/sdk/android/src/main/java/com/rudderstack/react/android/Utility.java

```diff
@@ -96,27 +99,59 @@ public static RudderTraits convertReadableMapToTraits(ReadableMap readableMap) {
-                for (int i = 0; i < externalIdsList.size(); i++) {
-                    Map<String, Object> externalId = (Map<String, Object>) externalIdsList.get(i);
-                    options.putExternalId((String) externalId.get("type"), (String) externalId.get("id"));
+        try {
+            Map<String, Object> optionsMap = convertReadableMapToMap(readableMap);
+            String[] externalIdKeys = {"externalId", "externalIds"};
+            for (String externalIdKey : externalIdKeys) {
+                if (readableMap.hasKey(externalIdKey) && readableMap.getType(externalIdKey) == ReadableType.Array) {
+                    List<Object> externalIdsList = convertReadableArrayToList(readableMap.getArray(externalIdKey));
+                    if (!isEmpty(externalIdsList)) {
+                        for (int i = 0; i < externalIdsList.size(); i++) {
+                            Map<String, Object> externalId = (Map<String, Object>) externalIdsList.get(i);
+                            if (isValidExternId(externalId)) {
+                                options.putExternalId(getString(externalId.get("type")), getString(externalId.get("id")));
+                            }
+                        }
+                        break;
+                    }
                 }
-                break;
             }
-        }
-        if (optionsMap.containsKey("integrations")) {
-            Map<String, Object> integrationsMap = (Map<String, Object>) optionsMap.get("integrations");
-            for (String key : integrationsMap.keySet()) {
-                options.putIntegration(key, (boolean) integrationsMap.get(key));
+            if (optionsMap.containsKey("integrations") && optionsMap.get("integrations") != null && optionsMap.get("integrations") instanceof Map) {
+                Map<String, Object> integrationsMap = (Map<String, Object>) optionsMap.get("integrations");
+                if (!isEmpty(integrationsMap)) {
+                    for (String key : integrationsMap.keySet()) {
+                        if (integrationsMap.get(key) != null && integrationsMap.get(key) instanceof Boolean) {
+                            options.putIntegration(key, getBoolean(integrationsMap.get(key)));
+                        }
+                    }
+                }
             }
+        } catch (Exception e) {
+            RudderLogger.logError("Exception occurred while handling options: " + e.getMessage());
         }
         return options;
     }
 
+    private static boolean isValidExternId(Map<String, Object> externalId) {
+        if (externalId == null || externalId.isEmpty()) return false;
+        if (!externalId.containsKey("type") || !externalId.containsKey("id")) return false;
+        return externalId.get("type") != null && externalId.get("id") != null;
+    }
+
+    public static String getString(Object obj) {
+        if (obj == null) return null;
```

Comment:
Why are we returning null if this method is supposed to return a string?

Also, I don't see the handling for `null` in the one place this is used.

Reply:
I've updated it to return an empty string.

`obj` could be `null`. Therefore I kept it to return null and not handling null was a mistake. Thanks for pointing to that.